### PR TITLE
[FrameworkBundle] Remove env var table from AboutCommand

### PR DESCRIPTION
**Pull request by tuqqu**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | -
| Deprecations? | -
| Tickets       | -
| License       | MIT
| Doc PR        | -

Fixed AboutCommand output by shortening environment variable value (replacing with `...`) to fit the screen width if the value is too long.

Right now all output is a mess if it does not fit in the terminal window.

### CODE REVIEW

**Comment by ro0NL** [comment]
it's considerable to use the new Console dumper (#28898) for env  var values, that would improve display as well.

The CliDumper has `setMaxStringWidth`, so if we can pass such a value (as calculated here) we can depend on existing building blocks :)

something like

```diff
diff --git a/Command/AboutCommand.php b/Command/AboutCommand.php
index b9fbe67..feee4e6 100644
--- a/Command/AboutCommand.php
+++ b/Command/AboutCommand.php
@@ -12,6 +12,7 @@
 namespace Symfony\Bundle\FrameworkBundle\Command;
 
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Dumper;
 use Symfony\Component\Console\Helper\Helper;
 use Symfony\Component\Console\Helper\TableSeparator;
 use Symfony\Component\Console\Input\InputInterface;
@@ -90,12 +91,13 @@ EOT
         ];
 
         if ($dotenv = self::getDotenvVars()) {
+            $dumper = new Dumper($io);
             $rows = array_merge($rows, [
                 new TableSeparator(),
                 ['<info>Environment (.env)</>'],
                 new TableSeparator(),
-            ], array_map(function ($value, $name) {
-                return [$name, $value];
+            ], array_map(function ($value, $name) use ($dumper) {
+                return [$name, $dumper($value, $maxLength)];
             }, $dotenv, array_keys($dotenv)));
         }
```

**Comment by tuqqu** [comment]
@ro0NL I'm not sure. `Dumper::__invoke()` does not allow to pass string's max length as the second parameter. Or are you suggesting to tweak its behaviour in this pr?

If so, `dump`'s output is standing out with it's bright colours and double quotes from all the rest. Do we really want this? 

If so, I could take a look into this.

**Comment by ro0NL** [comment]
>  Or are you suggesting to tweak its behaviour in this pr?

yep :)

> If so, dump's output is standing out with it's bright colours and double quotes from all the rest. Do we really want this?

IMHO yes, but perhaps post a screenshot to see what others think. Maybe it's too much indeed ... i was mostly triggered for being able to reuse `CliDumper::setMaxStringWidth` :)

**Comment by tuqqu** [comment]
@ro0NL I decided against using `Dumper` since we would rely on `CliDumper::setMaxStringWidth()` and it is part of `VarDumper` which is mostly a require-dev package, plus it adds a new dependency (on symfony/console). 

So I rewrote it a bit and it looks quite nice. I added a pic 
Screenshot 2019-12-03 at 23 33 12

Can you make a review?

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 1]
```diff
@@ -141,4 +144,28 @@ private static function getDotenvVars(): array
 
         return $vars;
     }
+
+    private static function getLongestColumnLength(array $dotenvVars): int
+    {
+        return max(18/* longest column text */, ...array_map('strlen', array_keys($dotenvVars)));
+    }
+
+    private static function adjustLength(string $value, int $maxLength): string
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: `truncate()`

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php [hunk 2]
```diff
@@ -141,4 +144,28 @@ private static function getDotenvVars(): array
 
         return $vars;
     }
+
+    private static function getLongestColumnLength(array $dotenvVars): int
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: imho it's fine to inline this above

**Review by ro0NL — APPROVED** [review]
> decided against using Dumper since we would rely on CliDumper::setMaxStringWidth() and it is part of VarDumper which is mostly a require-dev package

good point, i forgot we need to account for the fallback implementation as well.

LGTM in general :+1:

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php (hunk 1 above)
Comment (tuqqu) [reply src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: renamed

File: src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php (hunk 2 above)
Comment (tuqqu) [reply src/Symfony/Bundle/FrameworkBundle/Command/AboutCommand.php]: fixed

**Comment by tuqqu** [comment]
@ro0NL fixed all

**Comment by nicolas-grekas** [comment]
Not sure about this: while it improves the display, what if I *do* want to get the full value, eg to copy/paste it somewhere else?

**Comment by ro0NL** [comment]
from experience, i agree `about` can be over-cluttered with output. Typically access tokens & co completely break the table

image

the original intend was always "debugging"; give a hint about the values being used. Truncating does not defeat this purpose IMHO.

> what if I do want to get the full value

`bin/console debug:container --env-var APP_SECRET` :)

**Comment by nicolas-grekas** [comment]
Wouldn't it make more sense to just remove this table with env vars?
There is already `bin/console debug:container --env-vars` if one wants to display them.

**Comment by ro0NL** [comment]
hmpf... yes. There's a subtle difference.. (envs managed by .env vs envs used in DI), but i agree it's not really significant.

Alternatively, what about displaying `SYMFONY_DOTENV_VARS` comma separeated in `about`, to not loose this feature (it's also done in the webprofiler e.g.)

**Comment by ro0NL** [comment]
i tend to agree we should remove the Environment table in `about`. Actually the web profiler shows the server parameters used for that request (and highlights which are from .env)... this concept doesnt exists for the console.

The current table was a solution, prior to the later added `debug:container --env-vars`.

**Comment by tuqqu** [comment]
@ro0NL @nicolas-grekas ok, I see. I pushed a new commit which removes env var table completely.

**Review by ro0NL — APPROVED** [review]
probably best to update the PR title

**Review by javiereguiluz — APPROVED** [review]
Arthur thanks for this contribution (and congrats for being your first Symfony contribution!). And thanks for your patience during the review process.

**Comment by nicolas-grekas** [comment]
Thank you @tuqqu.

**Pull request merged**